components/affinda/sources/common/base.mjs

```diff
@@ -0,0 +1,125 @@
+import affinda from "../../affinda.app.mjs";
+import crypto from "crypto";
+
+export default {
+  props: {
+    affinda,
+    db: "$.service.db",
+    http: {
+      type: "$.interface.http",
+      customResponse: true,
+    },
+    organization: {
+      propDefinition: [
+        affinda,
+        "organization",
+      ],
+    },
+    webhookSignatureKey: {
+      type: "string",
+      label: "Webhook Signature Key",
+      description: "If provided, this source will verify if the webhook comes from Affinda. Locate it in your organization settings.",
+      secret: true,
+      optional: true,
+    },
+  },
+  methods: {
+    _getEventType() {
+      throw new Error("event type is not defined");
+    },
+    _getWebhookId() {
+      return this.db.get("webhookId");
+    },
+    _setWebhookId(id) {
+      this.db.set("webhookId", id);
+    },
+  },
+  hooks: {
+    async deploy() {
+      let offset = 0;
+      const limit = 300;
+      const documents = [];
+
+      while (true) {
+        const { results } = await this.affinda.listDocuments({
+          params: {
+            offset,
+            limit,
+          },
+        });
+        documents.push(...results);
+        if (results.length < limit) break;
+        offset += limit;
+      }
+
+      for (const { meta: document } of documents.reverse().slice(0, 50)) {
```

Comment:
why did you sliced it by 50 if you are limiting the request to at least 300?

Reply:
Slicing the last 50 historical events, limiting the HTTP request to 300 to make less requests. Does that make sense?